__tests__/classes/PaymentPreprocessor.js

```diff
@@ -0,0 +1,121 @@
+// eslint-disable-next-line no-unused-vars
+/* globals describe, beforeAll, afterAll, beforeEach, afterEach, test, expect */
+'use strict'
+
+const ah = require('../../test/ah-setup')
+const processPayment = require('../../classes/PaymentPreprocessor')
+const {generateMember, generatePayment} = require('../../test/generators')
+
+describe('payments preprocessor', async () => {
+  beforeAll(async () => {
+    await ah.start()
+    await ah.api.models.payment.destroy({where: {}, force: true})
+    await ah.api.models.member.destroy({where: {}, force: true})
+  })
+
+  afterAll(ah.stop)
+
+  afterEach(async () => {
+    await ah.api.models.payment.destroy({where: {}, force: true})
+    await ah.api.models.member.destroy({where: {}, force: true})
+  })
+
+  test('should add billing member in payment members when single payment is made', async () => {
+    await ah.api.models.member.create(generateMember())
+    const member2 = await ah.api.models.member.create(generateMember())
+    const payment = generatePayment({
+      billingMemberId: member2.id,
+      family: '0'
+    })
+    delete payment.members
+    const processed = await processPayment(payment)
+    expect(processed.members.length).toBe(1)
+    expect(processed.members[0]).toBe(member2.id)
+  })
+
+  test('should not add family members in payment members when single payment is made', async () => {
+    await ah.api.models.member.create(generateMember())
```

Comment:
you better have one test where you have 2 members and validate the processor is using the correct one, all the rest should not validate that